google/cloud/internal/filesystem.cc

```diff
@@ -170,6 +170,26 @@ std::uintmax_t file_size(std::string const& path,
   return static_cast<std::uintmax_t>(stat.st_size);
 }
 
+std::string PathAppend(std::string const& directory, std::string const& path) {
+#if _WIN32
+  auto constexpr kSeparator = '\\';
+#else
+  auto constexpr kSeparator = '/';
+#endif
+  if (path.empty()) return directory;
+  if (directory.empty()) return path;
+  if (directory.back() != kSeparator && path.front() != kSeparator) {
+    return directory + kSeparator + path;
+  }
+  if (directory.back() != kSeparator || path.front() != kSeparator) {
```

Comment:
`directory.back() != kSeparator` implies `path.front() != kSeparator` in this context.  I'm not sure whether making the redundant check renders the code more or less understandable.

Reply:
Yes, but `directory.back() == kSeparator` says nothing about whether `path.front()` matches or not?